### PR TITLE
106 functional tests

### PR DESCRIPTION
**Pull request by benred42**
Addresses #106.  Sets up functional testing with selenium and provides some initial tests for logging in and editing a scribble.

### CODE REVIEW

File: scribbler/tests/test_views.py [hunk 1]
```diff
@@ -392,3 +398,42 @@ def test_permission_required(self):
...
+        self.browser.find_element_by_name('submit').click()
+        scribble = self.browser.find_element_by_class_name("scribble-wrapper")
+        self.assertTrue(scribble)
+        editor = self.browser.find_element_by_id("scribbleEditorContainer")
+        self.assertTrue(editor)
+        scribble.click()
+        # The textarea in the editor is inside a div with height=0 which selenium views as not visible
+        # and therefore won't send keys to, so we have to give that div a non-zero height.
+        self.browser.execute_script("return document.getElementsByClassName('CodeMirror cm-s-default')[0].firstChild.style.height='1px';")
+        time.sleep(1)
```
Comment (benred42) [inline scribbler/tests/test_views.py]: It would be nice if we could disable JQuery animations instead of using a time.sleep() here.

**Comment by benred42** [comment]
It occurs to me that the build may be failing because the required libs are not included in the repo.

**Comment by mlavin** [comment]
Per our discussion yesterday, let's skip these on CI for now and address this along with rethinking how we packag/require these JS dependencies. That has been a long standing pain point.

File: scribbler/tests/test_views.py [hunk 2]
```diff
@@ -11,6 +11,16 @@
 from . import DaysLog
 from .base import ScribblerDataTestCase, Scribble
 
+import time
```
Comment (mlavin) [inline scribbler/tests/test_views.py]: Minor nit but can we try to keep these imports somewhat ordered: standard lib, Django, internal code in alpha order

File: scribbler/tests/urls.py [hunk 3]
```diff
@@ -1,16 +1,18 @@
...
-    from django.conf.urls import include, patterns, url, handler404, handler500
-except ImportError: # pragma: no cover
-    # Django 1.3
-    from django.conf.urls.defaults import include, patterns, url, handler404, handler500
 
+from scribbler.tests.views import testpage
 
-handler404 = 'scribbler.tests.urls.test_404'
-handler500 = 'scribbler.tests.urls.test_500'
+# try:
```
Comment (mlavin) [inline scribbler/tests/urls.py]: No sense in keeping this around commented out.

File: scribbler/tests/urls.py [hunk 4]
```diff
@@ -1,16 +1,18 @@
...
+#     # Django 1.4+
+#     from django.conf.urls import include, patterns, url, handler404, handler500
+# except ImportError: # pragma: no cover
+#     # Django 1.3
+#     from django.conf.urls.defaults import include, patterns, url, handler404, handler500
+#
+#
+# handler404 = 'scribbler.tests.urls.test_404'
```
Comment (mlavin) [inline scribbler/tests/urls.py]: Seems like we still want this part though, right?

File: scribbler/tests/views.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+from django.shortcuts import render, redirect
+from django.contrib.auth import login
+from django.contrib.auth.forms import AuthenticationForm
+
+
+def testpage(request):
```
Comment (mlavin) [inline scribbler/tests/views.py]: Could this just use the built-in `login` view from contrib auth? https://docs.djangoproject.com/en/1.8/topics/auth/default/#django.contrib.auth.views.login

File: runtests.py [hunk 6]
```diff
@@ -63,7 +69,11 @@
             # us skip creating migrations for the test models.
             'scribbler': 'scribbler.tests.migrations',
             'dayslog': 'dayslog.tests.migrations',
-        }
+        },
+        MEDIA_ROOT = '',
```
Comment (mlavin) [inline runtests.py]: Minor nit: these are inside the `settings.configure` call so they should not have spaces.

**Comment by mlavin** [comment]
:shipit:

**Pull request merged**